client/state/sites/updates/reducer.js

```diff
@@ -2,60 +2,61 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
+import { reduce } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { createReducer } from 'state/utils';
 
 import {
+	SITE_RECEIVE,
+	SITES_RECEIVE,
+	SITES_UPDATE,
 	SITE_UPDATES_RECEIVE,
 	SITE_UPDATES_REQUEST,
 	SITE_UPDATES_REQUEST_SUCCESS,
 	SITE_UPDATES_REQUEST_FAILURE,
-	SERIALIZE,
-	DESERIALIZE
 } from 'state/action-types';
 
 import { itemsSchema } from './schema';
 
-export const items = createReducer(
-	{},
-	{ [ SITE_UPDATES_RECEIVE ]: ( state, { siteId, updates } ) => Object.assign( {}, state, { [ siteId ]: updates } ) },
-	itemsSchema
-);
+const receiveUpdatesForSites = ( state, sites ) => {
+	return reduce( sites, ( memo, site ) => {
+		if ( memo === state ) {
+			memo = { ...state };
+		}
 
-export const requesting = ( state = false, { type } ) => {
-	switch ( type ) {
-		case SITE_UPDATES_REQUEST:
-		case SITE_UPDATES_REQUEST_SUCCESS:
-		case SITE_UPDATES_REQUEST_FAILURE:
-			return type === SITE_UPDATES_REQUEST;
+		if ( site.updates ) {
+			memo[ site.ID ] = site.updates;
+		}
 
-		case SERIALIZE:
-		case DESERIALIZE:
-			return false;
-	}
-
-	return state;
+		return memo;
+	}, state );
 };
 
-export const errors = ( state = false, { type } ) => {
-	switch ( type ) {
-		case SITE_UPDATES_REQUEST:
-		case SITE_UPDATES_REQUEST_SUCCESS:
-			return false;
-
-		case SITE_UPDATES_REQUEST_FAILURE:
-			return true;
+export const items = createReducer(
+	{},
+	{
+		[ SITE_UPDATES_RECEIVE ]: ( state, { siteId, updates } ) => Object.assign( {}, state, { [ siteId ]: updates } ),
+		[ SITE_RECEIVE ]: ( state, { site } ) => receiveUpdatesForSites( state, [ site ] ),
+		[ SITES_RECEIVE ]: ( state, { sites } ) => receiveUpdatesForSites( state, sites ),
+		[ SITES_UPDATE ]: ( state, { sites } ) => receiveUpdatesForSites( state, sites ),
+	},
+	itemsSchema
+);
 
-		case SERIALIZE:
-		case DESERIALIZE:
-			return false;
-	}
+export const requesting = createReducer( {}, {
+	[ SITE_UPDATES_REQUEST ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
+	[ SITE_UPDATES_REQUEST_SUCCESS ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+	[ SITE_UPDATES_REQUEST_FAILURE ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+} );
```

Comment:
These could maybe benefit from [`keyedReducer`](https://github.com/Automattic/wp-calypso/blob/50d131b90defd57c4382a0b97bb96caae3cc6517/client/state/utils.js#L77), but don't let that keep you from merging.